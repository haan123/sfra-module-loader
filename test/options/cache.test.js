/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('cache', () => {
    test('{Boolean} - `false` (Default)', async () => {
      const config = {
        loader: {
          test: /(js)/,
          options: {
            alias: {
              pluggin: 'pluggin',
              core: 'core'
            },
            cartridges: [
              'storefront',
              'pluggin',
              'core'
            ],
            cache: false
          }
        }
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { source } = module;

      expect(source).toMatchSnapshot();
    });

    test('{Boolean} - `true`', async () => {
      const config = {
        loader: {
          test: /(js)/,
          options: {
            alias: {
              pluggin: 'pluggin',
              core: 'core'
            },
            cartridges: [
              'storefront',
              'pluggin',
              'core'
            ],
            cache: true
          }
        }
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { source } = module;

      expect(source).toMatchSnapshot();
    });
  });
});
