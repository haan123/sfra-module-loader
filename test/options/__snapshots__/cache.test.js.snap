// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options cache {Boolean} - \`false\` (Default) 1`] = `
"export default function () {
  return 'core';
};
"
`;

exports[`Options cache {Boolean} - \`true\` 1`] = `
"export default function () {
  return 'core';
};
"
`;
